### ISSUE TITLE
[FEATURE]: Improve semantic HTML & accessibility for NavBar and Hero components

### ISSUE THREAD

**Issue by aliceafanasieva** [fdnd-agency/embassyofthefreemind#244]
## Task description

Goal is to improve the NavBar and Hero components in terms of:

- semantic HTML
- Accessibility (WCAG, A11y principles)
- Performance (minimizing unnecessary JavaScript and ARIA)

Use semantic HTML elements as the primary solution, apply ARIA only when HTML semantics are not sufficient.

### Lighthouse Test

The current score on Accessibility and Performance should be improved:

#### Performance: 74

<img width="1003" height="950" alt="Image" src="https://github.com/user-attachments/assets/6e1cfca3-e66e-4e13-992f-9469b1b6e50f" />

#### Accessibility: 84

<img width="1003" height="950" alt="Image" src="https://github.com/user-attachments/assets/9881fb56-179c-40a0-ba05-2e77e42acb45" />
